### ISSUE TITLE
toolchain: get_openblas_arch.sh fails on modern Ryzens

### ISSUE THREAD

**Issue by michaellass** [cp2k/cp2k#1731]
I get the following output when trying to install the toolchain via `tools/toolchain/install_cp2k_toolchain.sh`:

```
MPI is detected and it appears to be OpenMPI
Compiling with 16 processes.
==================== Finding GCC from system paths ====================
path to gcc is  /usr/bin/gcc
path to g++ is  /usr/bin/g++
path to gfortran is  /usr/bin/gfortran
Found include directory /usr/include
Found lib directory /usr/lib
Step gcc took 0.00 seconds.
==================== Getting proc arch info using OpenBLAS tools ====================
SSL_INIT
OpenBLAS-0.3.10.tar.gz: OK
Checksum of OpenBLAS-0.3.10.tar.gz Ok
getarch_2nd.c: In function 'main':
getarch_2nd.c:14:35: error: 'SGEMM_DEFAULT_UNROLL_M' undeclared (first use in this function); did you mean 'SHGEMM_DEFAULT_UNROLL_M'?
   14 |     printf("SGEMM_UNROLL_M=%d\n", SGEMM_DEFAULT_UNROLL_M);
      |                                   ^~~~~~~~~~~~~~~~~~~~~~
      |                                   SHGEMM_DEFAULT_UNROLL_M
getarch_2nd.c:14:35: note: each undeclared identifier is reported only once for each function it appears in
getarch_2nd.c:15:35: error: 'SGEMM_DEFAULT_UNROLL_N' undeclared (first use in this function); did you mean 'QGEMM_DEFAULT_UNROLL_N'?
   15 |     printf("SGEMM_UNROLL_N=%d\n", SGEMM_DEFAULT_UNROLL_N);
      |                                   ^~~~~~~~~~~~~~~~~~~~~~
      |                                   QGEMM_DEFAULT_UNROLL_N
getarch_2nd.c:16:35: error: 'DGEMM_DEFAULT_UNROLL_M' undeclared (first use in this function); did you mean 'XGEMM_DEFAULT_UNROLL_M'?
   16 |     printf("DGEMM_UNROLL_M=%d\n", DGEMM_DEFAULT_UNROLL_M);
      |                                   ^~~~~~~~~~~~~~~~~~~~~~
      |                                   XGEMM_DEFAULT_UNROLL_M
getarch_2nd.c:17:35: error: 'DGEMM_DEFAULT_UNROLL_N' undeclared (first use in this function); did you mean 'QGEMM_DEFAULT_UNROLL_N'?
   17 |     printf("DGEMM_UNROLL_N=%d\n", DGEMM_DEFAULT_UNROLL_N);
      |                                   ^~~~~~~~~~~~~~~~~~~~~~
      |                                   QGEMM_DEFAULT_UNROLL_N
getarch_2nd.c:21:35: error: 'CGEMM_DEFAULT_UNROLL_M' undeclared (first use in this function); did you mean 'XGEMM_DEFAULT_UNROLL_M'?
   21 |     printf("CGEMM_UNROLL_M=%d\n", CGEMM_DEFAULT_UNROLL_M);
      |                                   ^~~~~~~~~~~~~~~~~~~~~~
      |                                   XGEMM_DEFAULT_UNROLL_M
getarch_2nd.c:22:35: error: 'CGEMM_DEFAULT_UNROLL_N' undeclared (first use in this function); did you mean 'QGEMM_DEFAULT_UNROLL_N'?
   22 |     printf("CGEMM_UNROLL_N=%d\n", CGEMM_DEFAULT_UNROLL_N);
      |                                   ^~~~~~~~~~~~~~~~~~~~~~
      |                                   QGEMM_DEFAULT_UNROLL_N
getarch_2nd.c:23:35: error: 'ZGEMM_DEFAULT_UNROLL_M' undeclared (first use in this function); did you mean 'XGEMM_DEFAULT_UNROLL_M'?
   23 |     printf("ZGEMM_UNROLL_M=%d\n", ZGEMM_DEFAULT_UNROLL_M);
      |                                   ^~~~~~~~~~~~~~~~~~~~~~
      |                                   XGEMM_DEFAULT_UNROLL_M
getarch_2nd.c:24:35: error: 'ZGEMM_DEFAULT_UNROLL_N' undeclared (first use in this function); did you mean 'QGEMM_DEFAULT_UNROLL_N'?
   24 |     printf("ZGEMM_UNROLL_N=%d\n", ZGEMM_DEFAULT_UNROLL_N);
      |                                   ^~~~~~~~~~~~~~~~~~~~~~
      |                                   QGEMM_DEFAULT_UNROLL_N
getarch_2nd.c:71:50: error: 'SGEMM_DEFAULT_Q' undeclared (first use in this function); did you mean 'SHGEMM_DEFAULT_Q'?
   71 |     printf("#define SLOCAL_BUFFER_SIZE\t%ld\n", (SGEMM_DEFAULT_Q * SGEMM_DEFAULT_UNROLL_N * 4 * 1 *  sizeof(float)));
      |                                                  ^~~~~~~~~~~~~~~
      |                                                  SHGEMM_DEFAULT_Q
getarch_2nd.c:72:50: error: 'DGEMM_DEFAULT_Q' undeclared (first use in this function); did you mean 'SHGEMM_DEFAULT_Q'?
   72 |     printf("#define DLOCAL_BUFFER_SIZE\t%ld\n", (DGEMM_DEFAULT_Q * DGEMM_DEFAULT_UNROLL_N * 2 * 1 *  sizeof(double)));
      |                                                  ^~~~~~~~~~~~~~~
      |                                                  SHGEMM_DEFAULT_Q
getarch_2nd.c:73:50: error: 'CGEMM_DEFAULT_Q' undeclared (first use in this function); did you mean 'SHGEMM_DEFAULT_Q'?
   73 |     printf("#define CLOCAL_BUFFER_SIZE\t%ld\n", (CGEMM_DEFAULT_Q * CGEMM_DEFAULT_UNROLL_N * 4 * 2 *  sizeof(float)));
      |                                                  ^~~~~~~~~~~~~~~
      |                                                  SHGEMM_DEFAULT_Q
getarch_2nd.c:74:50: error: 'ZGEMM_DEFAULT_Q' undeclared (first use in this function); did you mean 'SHGEMM_DEFAULT_Q'?
   74 |     printf("#define ZLOCAL_BUFFER_SIZE\t%ld\n", (ZGEMM_DEFAULT_Q * ZGEMM_DEFAULT_UNROLL_N * 2 * 2 *  sizeof(double)));
      |                                                  ^~~~~~~~~~~~~~~
      |                                                  SHGEMM_DEFAULT_Q
make: *** [Makefile.prebuild:70: getarch_2nd] Error 1
OpenBLAS detected LIBCORE = unknown
OpenBLAS detected ARCH    = x86_64
==================== Installing CMake ====================
[...]
```

Turns out that OpenBLAS 0.3.10 as included in the toolchain does not support my AMD Renoir CPU yet. It's fixed in OpenBLAS 0.3.11 by https://github.com/xianyi/OpenBLAS/commit/200f5c44cc14f356d7dba6af257044016a0573da.

Would it make sense to update the OpenBLAS version in the toolchain before the 9.1 release to support more recent AMD hardware?

**Comment by michaellass** [comment]
Dirty patch without updating OpenBLAS:
```diff
diff --git a/tools/toolchain/scripts/get_openblas_arch.sh b/tools/toolchain/scripts/get_openblas_arch.sh
index 59b2afc54..52aafdae9 100755
--- a/tools/toolchain/scripts/get_openblas_arch.sh
+++ b/tools/toolchain/scripts/get_openblas_arch.sh
@@ -51,6 +51,7 @@ openblas_conf="${openblas_dir}/Makefile.conf"
 # try find Makefile.config, if not then generate one with make lapack_prebuild
 if ! [ -f "$openblas_conf" ]; then
   cd "$openblas_dir"
+  curl https://github.com/xianyi/OpenBLAS/commit/200f5c44cc14f356d7dba6af257044016a0573da.patch | patch -p1
   make lapack_prebuild
   cd ..
 fi
diff --git a/tools/toolchain/scripts/stage2/install_openblas.sh b/tools/toolchain/scripts/stage2/install_openblas.sh
index 5e7a18c0f..ee3cd9195 100755
--- a/tools/toolchain/scripts/stage2/install_openblas.sh
+++ b/tools/toolchain/scripts/stage2/install_openblas.sh
@@ -48,6 +48,8 @@ case "$with_openblas" in
       tar -zxf ${openblas_pkg}
       cd OpenBLAS-${openblas_ver}
 
+      curl https://github.com/xianyi/OpenBLAS/commit/200f5c44cc14f356d7dba6af257044016a0573da.patch | patch -p1
+
       # First attempt to make openblas using auto detected
       # TARGET, if this fails, then make with forced
       # TARGET=NEHALEM
```

Makes both `Getting proc arch info using OpenBLAS tools` as well as `Installing OpenBLAS` work:

```
==================== Getting proc arch info using OpenBLAS tools ====================
SSL_INIT
OpenBLAS-0.3.10.tar.gz: OK
Checksum of OpenBLAS-0.3.10.tar.gz Ok
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2095  100  2095    0     0   5432      0 --:--:-- --:--:-- --:--:--  5441
patching file cpuid_x86.c
Hunk #1 succeeded at 1443 (offset -11 lines).
Hunk #2 succeeded at 1497 (offset -11 lines).
Hunk #3 succeeded at 1508 (offset -11 lines).
Hunk #4 succeeded at 2096 (offset -24 lines).
Hunk #5 succeeded at 2124 (offset -24 lines).
OpenBLAS detected LIBCORE = zen
OpenBLAS detected ARCH    = x86_64
```

```
==================== Installing OpenBLAS ====================
OpenBLAS-0.3.10.tar.gz is found
Installing from scratch into /home/bevan/scratch/git/cp2k/tools/toolchain/install/openblas-0.3.10
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2095  100  2095    0     0   7010      0 --:--:-- --:--:-- --:--:--  7006
patching file cpuid_x86.c
Hunk #1 succeeded at 1443 (offset -11 lines).
Hunk #2 succeeded at 1497 (offset -11 lines).
Hunk #3 succeeded at 1508 (offset -11 lines).
Hunk #4 succeeded at 2096 (offset -24 lines).
Hunk #5 succeeded at 2124 (offset -24 lines).
Step openblas took 372,00 seconds.
```

**Comment by mkrack** [comment]
@michaellass Many thanks for reporting the issue. I think we should update the toolchain to the latest [OpenBLAS release 0.3.18.](https://github.com/xianyi/OpenBLAS/releases/tag/v0.3.18) I will look into that.

**Comment by mkrack** [comment]
Resolved via #1732
